source/js/RecipeExpand.js

```diff
@@ -0,0 +1,496 @@
+   * @param {String} key the key that you are looking for in the object
+   * @returns {*} the value of the found key
+   */
+  function searchForKey(object, key) {
+    var value;
+    Object.keys(object).some(function (k) {
+      if (k === key) {
+        value = object[k];
+        return true;
+      }
+      if (object[k] && typeof object[k] === 'object') {
+        value = searchForKey(object[k], key);
+        return value !== undefined;
+      }
+    });
+    return value;
+  }
+  
+  /**
+   * Extract the title of the recipe from the given recipe schema JSON obejct
+   * @param {Object} data Raw recipe JSON to find the image of
+   * @returns {String} If found, returns the recipe title
+   */
+  function getTitle(data) {
+    if (data.name) return data.name;
+    if (data['@graph']) {
+      for (let i = 0; i < data['@graph'].length; i++) {
+        if (data['@graph'][i]['@type'] == 'Recipe') {
+          if (data['@graph'][i]['name']) return data['@graph'][i]['name'];
+        };
+      }
+    }
+    return null;
+  }
+  
+  /**
+   * Extract the yield of the recipe from the given recipe schema JSON obejct
+   * @param {Object} data Raw recipe JSON to find the image of
+   * @returns {String} If found, returns the recipe yield
+   */
+  function getYield(data) {
+    if (data.recipeYield) return data.recipeYield;
+    if (data['@graph']) {
+      for (let i = 0; i < data['@graph'].length; i++) {
+        if (data['@graph'][i]['@type'] == 'Recipe') {
+          if (data['@graph'][i]['recipeYield']) {
+            if (Array.isArray(data['@graph'][i]['recipeYield'])) {
+              return data['@graph'][i]['recipeYield'][0];
+            } else if (typeof data['@graph'][i]['recipeYield'] == 'string') {
+              return data['@graph'][i]['recipeYield'];
+            }
+          }
+        }
+      }
+    }
+    return null;
+  }
+  
+  /**
+   * Extract the categories of the recipe from the given recipe schema JSON obejct
+   * @param {Object} data Raw recipe JSON to find the image of
+   * @returns {String} If found, returns the recipe categories as a string
+   */
+  function getCategories(data) {
+    let categories = null;
+    if (data.recipeCategory) {
+      categories = data.recipeCategory
+    } else if (data['@graph']) {
+      for (let i = 0; i < data['@graph'].length; i++) {
+        if (data['@graph'][i]['@type'] == 'Recipe') {
+          if (data['@graph'][i]['recipeCategory']) {
+            categories = data['@graph'][i]['recipeCategory'];
+          }
+        };
+      }
+    }
+    if (Array.isArray(categories)) categories = categories.join(', ');
+    return categories.toLowerCase();
+  }
+  
+  /**
+   * Extract the description of the recipe from the given recipe schema JSON obejct
+   * @param {Object} data Raw recipe JSON to find the image of
+   * @returns {String} If found, returns the recipe description
+   */
+  function getDescription(data) {
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 0 spaces but found 2.](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8447291)
```suggestion
function getDescription(data) {
```